src/main/java/mate/academy/bookstore/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,98 @@
+package mate.academy.bookstore.service.order;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstore.dto.order.OrderResponseDto;
+import mate.academy.bookstore.dto.order.ShippingAddressRequestDto;
+import mate.academy.bookstore.dto.order.StatusRequestDto;
+import mate.academy.bookstore.dto.orderitem.OrderItemDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.CartItemMapper;
+import mate.academy.bookstore.mapper.OrderItemMapper;
+import mate.academy.bookstore.mapper.OrderMapper;
+import mate.academy.bookstore.model.Order;
+import mate.academy.bookstore.model.OrderItem;
+import mate.academy.bookstore.repository.order.OrderRepository;
+import mate.academy.bookstore.repository.orderitem.OrderItemRepository;
+import mate.academy.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import mate.academy.bookstore.repository.user.UserRepository;
+import mate.academy.bookstore.service.shoppingcart.ShoppingCartService;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartRepository cartRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartService shoppingCartService;
+
+    @Override
+    public void createOrder(Authentication authentication, ShippingAddressRequestDto requestDto) {
```

Comment:
method became too bulky and hard to read 
try to extract some logic to private methods like this

        User user = fetchUserByEmail(authentication.getName());
        Set<OrderItem> orderItems = fetchOrderItems(authentication.getName());
        Order order = createNewOrder(user, requestDto.getShippingAddress(), orderItems);
        saveOrderWithItems(order);
        shoppingCartService.clearShoppingCart(authentication);